scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+	{
+		// We don't clear the glyphs any more or at least until the whole page has been rendred glyphs.clear();
+		return;
+	}
+
+	//FIXME: This is redundant, should be using the framework not the first glyph.
+	//const PdfGlyph& first_glyph = activeTextRegion.glyphs[activeTextRegion.textRegionLines[0].glyphIndex];// (*i);
+	// TODO: Use the framework for this, not currently supported
+	/*
+	int render_mode = first_glyph.render_mode;
+	// Ignore invisible characters
+	if (render_mode == 3) {
+		// We don't clear the glyphs any more or at least until the whole page has been rendred //_glyphs.clear();
+		return;
+	}
+	*/
+	//FIXME: Use the framework for positioning not the first glyph
+	qreal xCoor = m_doc->currentPage()->xOffset() + activeTextRegion->textRegioBasenOrigin.x();
+	qreal yCoor = m_doc->currentPage()->initialHeight() - (m_doc->currentPage()->yOffset() + (double)activeTextRegion->textRegioBasenOrigin.y() + activeTextRegion->lineSpacing); // don't know if y is top down or bottom up
+	qreal  lineWidth = 0.0;
+#ifdef DEBUG_TEXT_IMPORT
+	qDebug() << "rendering new frame at:" << xCoor << "," << yCoor << " With lineheight of: " << activeTextRegion->lineSpacing << "Height:" << activeTextRegion->maxHeight << " Width:" << activeTextRegion->maxWidth;
+#endif
+	/* colours don't get reset to CommonStrings::None often enough.*/
+	int z = m_doc->itemAdd(PageItem::TextFrame, PageItem::Rectangle, xCoor, yCoor, 40, 40, 0, CommonStrings::None, CommonStrings::None /* this->CurrColorStroke*/);//, PageItem::ItemKind::InlineItem);
+	PageItem* textNode = m_doc->Items->at(z);
+
+	ParagraphStyle& pStyle = (ParagraphStyle&)textNode->itemText.defaultStyle();
+	// set some hackish parameters up at first, line spacing can be calculated from the cursor position changes
+	pStyle.setLineSpacingMode(pStyle.AutomaticLineSpacing);
+	pStyle.setHyphenationMode(pStyle.AutomaticHyphenation);
+
+	// TODO: Implement thease using the framework
+	finishItem(textNode);
+	// FIXME: Implement thease using the framework
+	//_setFillAndStrokeForPdf(state, text_node);
+	//FIXME: Here's some dummy code for now with sednsible defaults, looks like state wasn't even needed
+	if (true)
+	{
+		textNode->ClipEdited = true;
+		textNode->FrameType = 3;
+		textNode->setLineEnd(PLineEnd);
+		textNode->setLineJoin(PLineJoin);
+		textNode->setTextFlowMode(PageItem::TextFlowDisabled);
+		//textNode->setFillTransparency(1.0);
+		textNode->setLineTransparency(1.0); // this ssets the transparency of the textbox border and we don't want to see it			
+		textNode->setFillColor(CommonStrings::None);
+		textNode->setLineColor(CommonStrings::None);
+		textNode->setLineWidth(0);//line  width doesn't effect drawing text, it creates a bounding box state->getTransformedLineWidth());
+		textNode->setFillShade(CurrFillShade);
+	}
+
+	// Set text matrix... This need to be done so that the globaal world view that we rite out glyphs to is transformed correctly by the context matrix for each glyph, possibly anyhow.

```

Comment:
when leaving in commented out code, you should add your name, the date and the conditions that are necessary for removing / activating the code.

Reply:
ok